apps/veyfi/components/LockTab.tsx

```diff
@@ -0,0 +1,174 @@
+		if (!votingEscrow || !userAddress) {
+			return;
+		}
+		approveLock(web3Provider, userAddress, votingEscrow.token, votingEscrow.address);
+	};
+	
+	const executeLock = (): void => {
+		if (!votingEscrow  || !userAddress) {
+			return;
+		}
+		lock(web3Provider, userAddress, votingEscrow.address, lockAmount.raw, toSeconds(unlockTime));
+	};
+	
+	const executeIncreaseLockAmount = (): void => {
+		if (!votingEscrow  || !userAddress) {
+			return;
+		}
+		increaseLockAmount(web3Provider, userAddress, votingEscrow.address, lockAmount.raw);
+	};
+
+	const txAction = !isApproved
+		? {
+			label: 'Approve',
+			onAction: executeApprove,
+			status: approveLockStatus.loading,
+			disabled: !isActive || !isValidNetwork || isApproved || isLoadingVotingEscrow
+		}
+		: hasLockedAmount
+			? {
+				label: 'Lock',
+				onAction: executeIncreaseLockAmount,
+				status: increaseLockAmountStatus.loading,
+				disabled: !isActive || !isValidNetwork || !isApproved || !isValidLockAmount || !isValidLockTime || isLoadingVotingEscrow
+			}
+			: {
+				label: 'Lock',
+				onAction: executeLock,
+				status: lockStatus.loading,
+				disabled: !isActive || !isValidNetwork || !isApproved || !isValidLockAmount || !isValidLockTime || isLoadingVotingEscrow
+			};
+
+	return ( 
+		<div className={'grid grid-cols-1 gap-6 md:grid-cols-2 md:gap-16'}>
+			<div className={'col-span-1 w-full'}>
+				<h2 className={'m-0 text-2xl font-bold'}>
+					{"YFI holders, time to Lock' ‘N Load"}
+				</h2>
+				<div className={'mt-6 text-neutral-600'} >
+					<p >{'Lock your YFI for veYFI to take part in Yearn governance.'}</p>
+					<br />
+					<p>{'Please note, governance is currently the only use for veYFI until the full platform launches ‘soon’. Stay tuned anon.'}</p>
+				</div>
+			</div>
+
+			<div className={'col-span-1 grid w-full gap-6'}>
+				<div className={'mt-0 grid grid-cols-1 gap-6 md:mt-14 md:grid-cols-2'}>
+					<AmountInput
+						label={'YFI'}
+						amount={lockAmount.normalized}
+						onAmountChange={(amount): unknown => set_lockAmount(toNormalizedBN(amount, 18))}
+						maxAmount={tokenBalance.normalized > 0 ? tokenBalance.normalized.toFixed(18) : ''}
+						legend={`Available: ${formatAmount(tokenBalance.normalized, 4)} YFI`}
+						error={lockAmountError}
+					/>
+					<AmountInput
+						label={'Current lock period (weeks)'}
+						amount={toTime(lockTime) === 0 ? '' : Math.floor(toTime(lockTime)).toString()}
+						onAmountChange={set_lockTime}
+						maxAmount={(MAX_LOCK_TIME + 1).toString()}
+						disabled={hasLockedAmount}
+						legend={'min 1'}
+						error={lockTimeError}
+					/>
+				</div>
+				<div className={'grid grid-cols-1 gap-6 md:grid-cols-2'}>
+					<AmountInput
+						label={'Total veYFI'}
+						amount={formatUnits(votingPower, 18)}
+						disabled
+					/>
+					<Button
+						className={'w-full md:mt-7'}
+						onClick={txAction.onAction}
+						isDisabled={txAction.disabled || txAction.status}
+						isBusy={txAction.status}
```

Comment:
This `txAction.status` sounds a bit mysterious as a boolean, so if `status` is `true` then the button is disabled? A bit confusing, also as a boolean probably should be renamed to `isStatus` or a better name :)

Reply:
I'd actually rename `disabled` to `isDisabled` and `status` to `isLoading`